lxd/images.go

```diff
@@ -316,20 +318,23 @@ func getImageMetadata(fname string) (*imageMetadata, error) {
 	default:
 		args = append(args, "-Jxf")
 	}
-	args = append(args, fname, "metadata.yaml")
+	args = append(args, fname, metadataName)
+
+	shared.Debugf("Extracting tarball using command: tar %s", strings.Join(args, " "))
 
 	// read the metadata.yaml
-	output, err := exec.Command("tar", args...).Output()
+	output, err := exec.Command("tar", args...).CombinedOutput()
 
 	if err != nil {
-		return nil, fmt.Errorf("Could not get image metadata: %v", err)
+		outputLines := strings.Split(string(output), "\n")
+		return nil, fmt.Errorf("Could not extract image %s from tar: %v (%s)", metadataName, err, outputLines[0])
```

Comment:
Either "image metadata" or just "extract metadata", but this isn't extracting an image, so the error message is misleading.